### PR TITLE
Group right operator

### PR DESCRIPTION
**Pull request by Steph4104**
Add the function to check if the user is an admin or the creator to the content.

issue #1565

### CODE REVIEW

**Comment by ethanWallace** [comment]
Discussion creators see this on replies.
group owner

File: mod/groups/start.php [hunk 1]
```diff
@@ -1255,21 +1255,23 @@ function discussion_reply_menu_setup($hook, $type, $return, $params) {
 
 	// Allow discussion topic owner, group owner and admins to edit and delete
 	if ($reply->canEdit() && !elgg_in_context('activity')) {
-		$return[] = ElggMenuItem::factory(array(
-			'name' => 'edit',
-			'text' => elgg_echo('edit'),
-			'href' => "discussion/reply/edit/{$reply->guid}",
-			'priority' => 150,
-		));
+		if ($entity['owner_guid'] == $user['guid'] || elgg_is_admin_logged_in()){
```
Comment (nschonni) [inline mod/groups/start.php]: `$entity` isn't declared in this scope

File: mod/wet4/start.php [hunk 2]
```diff
@@ -992,11 +992,12 @@ function wet4_elgg_entity_menu_setup($hook, $type, $return, $params) {
...
                 $return[] = ElggMenuItem::factory($options);
             }
+    $user = elgg_get_logged_in_user_entity();
 	if ($entity->canEdit() && $handler) {
-		// edit link
+         if ($entity['owner_guid'] == $user['guid'] || elgg_is_admin_logged_in()){
 
         //checks so the edit icon is not placed on incorrect entities
-        if($handler != 'group_operators'){
+       // if($handler != 'group_operators'){
```
Comment (piet0024) [inline mod/wet4/start.php]: clean up? does the if on line 997 replace this one?

File: engine/lib/navigation.php [hunk 3]
```diff
@@ -471,32 +471,35 @@ function _elgg_entity_menu_setup($hook, $type, $return, $params) {
...
-			'text' => elgg_echo('edit'),
-			'title' => elgg_echo('edit:this'),
-			'href' => "$handler/edit/{$entity->getGUID()}",
-			'priority' => 200,
-		);
-		$return[] = \ElggMenuItem::factory($options);
-
-		if(elgg_is_logged_in()){
-			// delete link
+		if ($entity['owner_guid'] == $user['guid'] || elgg_is_admin_logged_in()){
```
[the remark this answers is not in the document]
Comment (Steph4104) [reply engine/lib/navigation.php]: If we dont replace the engine, It will show up.

[the remark this answers is not in the document]
Comment (markwooff) [reply engine/lib/navigation.php]: I briefly looked into this before and removing the line at https://github.com/gctools-outilsgc/gcconnex/blob/gcconnex/mod/groups/start.php#L1158 should have the same effect - no?

**Pull request merged**